components/org.wso2.carbon.extension.identity.sso.jwt/src/main/java/org/wso2/carbon/identity/sso/jwt/factory/JWTInboundRequestFactory.java

```diff
@@ -0,0 +1,92 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.sso.jwt.factory;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.FrameworkClientException;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.HttpIdentityRequestFactory;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.IdentityRequest;
+import org.wso2.carbon.identity.sso.jwt.message.JWTInboundRequest;
+import org.wso2.carbon.identity.sso.jwt.util.JWTInboundConstants;
+import org.wso2.carbon.identity.sso.jwt.util.JWTInboundUtil;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This class represents a factory for JWT IdentityRequest instances which will be passed to the framework for
+ * authentication.
+ *
+ * Essentially, the conversion of the protocol-specific HTTP request to the framework-understood IdentityRequest takes
+ * place here.
+ *
+ * This class is extensible either by implementing the HttpIdentityRequestFactory or extending the
+ * JWTInboundRequestFactory. The customized class should be defined under
+ * <JWTSSO><RequestFactory></RequestFactory></JWTSSO> in identity.xml. Otherwise JWTInboundRequestFactory will be
+ * used as the default request processor.
+ */
+public class JWTInboundRequestFactory extends HttpIdentityRequestFactory {
+
+    private static Log log = LogFactory.getLog(JWTInboundRequestFactory.class);
+
+    /**
+     * Checks whether or not an incoming request hitting the "/identity" servlet should be handled by this
+     * particular JWT IdentityRequest factory.
+     *
+     * @param request  the request parameter coming from the servlet
+     * @param response the response parameter coming from the servlet
+     * @return true if the request is of a type which can be handled by this particular IdentityRequest factory
+     */
+    @Override
+    public boolean canHandle(HttpServletRequest request, HttpServletResponse response) {
+        //Return true if the incoming request to the identity servlet in the form of : "/identity/BASE_PATH".
+        String requestUri = request.getRequestURI();
+        if (log.isDebugEnabled()) {
+            log.debug("Request URI: " + JWTInboundUtil.neutralize(requestUri));
+        }
+        if (StringUtils.isNotBlank(requestUri) && requestUri.contains(JWTInboundConstants.BASE_PATH)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Request URI contains the base path: " + JWTInboundConstants.BASE_PATH);
+            }
+            log.info("Handling authentication request through JWT Inbound Authenticator.");
```

Comment:
Lets make this a debug too. coz every time a user authenticates this will be logged in the terminal. And if there is a huge number of users there will be a load of info logs in the terminal

Reply:
Fixed